lib/wibox/hierarchy.lua

```diff
@@ -51,6 +51,15 @@ local function hierarchy_new(redraw_callback, layout_callback, callback_arg)
         end
         layout_callback(result, callback_arg)
     end
+    function result._emit_recursive(_, name, ...)
+        local cur = result
```

Comment:
Didn't you previously have this code in some other PR? My comments from there still apply (I'd make use of the `self` argument instead of ignoring it).